src/app/page/component/cyoa/card/card.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Observable } from 'rxjs';
+import { PageService } from 'src/app/page/service/page-service.service';
+import {
+  CYOAPageCard,
+  Content
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+
+@Component({
+  selector: 'app-page-cyoa-card',
+  templateUrl: './card.component.html',
+  styleUrls: ['./card.component.css']
+})
+export class CyoaCardComponent implements OnChanges {
+  @Input() card: CYOAPageCard;
+  @Input() totalCards: number;
+
+  ready: boolean;
+  cardPosition: number;
+  content: Array<Content>;
+  dir$: Observable<string>;
+  isForm$: Observable<boolean>;
+  isModal$: Observable<boolean>;
+  isFirstPage$: Observable<boolean>;
+
+  constructor(private pageService: PageService) {
+    this.dir$ = this.pageService.pageDir$;
+    this.isForm$ = this.pageService.isForm$;
+    this.isModal$ = this.pageService.isModal$;
+    this.isFirstPage$ = this.pageService.isFirstPage$;
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'card': {
+            if (
+              !changes['card'].previousValue ||
+              changes['card'].currentValue !== changes['card'].previousValue
+            ) {
+              this.ready = false;
+              this.cardPosition = 0;
+              this.content = [];
+              this.init();
+            }
+          }
+        }
```

Comment:
As we're only using one clause for the switch statement, it might be better to just switch it to a If statement.
I also know we prob do this on a lot of other components.